chia/rpc/wallet_rpc_api.py

```diff
@@ -1233,15 +1234,39 @@ async def verify_signature(self, request) -> EndpointResult:
         :param request:
         :return:
         """
+        input_message: str = request["message"]
+        signing_mode_str: Optional[str] = request.get("signing_mode")
+        # Default to BLS_MESSAGE_AUGMENTATION_HEX_INPUT as this RPC was originally designed to verify
+        # signatures made by `chia keys sign`, which uses BLS_MESSAGE_AUGMENTATION_HEX_INPUT
+        signing_mode = (
+            SigningMode(signing_mode_str)
+            if signing_mode_str is not None
+            else SigningMode.BLS_MESSAGE_AUGMENTATION_HEX_INPUT
+        )
+        if signing_mode == SigningMode.CHIP_0002:
+            # CHIP-0002 message signatures are made over the tree hash of:
+            #   ("Chia Signed Message", message)
+            message_to_verify: bytes = Program.to((CHIP_0002_SIGN_MESSAGE_PREFIX, input_message)).get_tree_hash()
+        elif signing_mode == SigningMode.BLS_MESSAGE_AUGMENTATION_HEX_INPUT:
+            # Message is expected to be a hex string
+            message_to_verify = hexstr_to_bytes(input_message)
+        elif signing_mode == SigningMode.BLS_MESSAGE_AUGMENTATION_UTF8_INPUT:
+            # Message is expected to be a UTF-8 string
+            message_to_verify = bytes(input_message, "utf-8")
```

Comment:
I would have expected something to complain about the maybe not defined `message_to_verify` variable.  I think an else here with an exception would be good to indicate that we are not handling a signing mode that we have defined.

I played around a bit and while I was able to trigger both mypy and flake8 to complain when I removed all references to creating the variable, they seem to be doing a really bad job of detecting this...  `:[`

Reply:
Added an else case